BeeSwift/Gallery/GalleryViewController.swift

```diff
@@ -27,14 +27,21 @@ class GalleryViewController: UIViewController, UICollectionViewDelegateFlowLayou
     let cellReuseIdentifier = "Cell"
     var deadbeatView = UIView()
     var outofdateView = UIView()
-    let noGoalsLabel = BSLabel()
     let outofdateLabel = BSLabel()
     let searchBar = UISearchBar()
     var lastUpdated: Date?
     let maxSearchBarHeight: Int = 50
     
-    var goals : [Goal] = []
-    var filteredGoals : [Goal] = []
+    var goals : [Goal] = [] {
+        didSet {
```

Comment:
[Do] Move this logic (and filteredGoals.didSet) to `didUpdateGoals` which contains all the other logic/view invalidation which is run when the set of goals update.